Plugins~/Src/MeshSyncClientBlender/msblenUtils.cpp

```diff
@@ -73,19 +86,24 @@ bool visible_in_viewport(const Object *obj)
     return !bl::BObject(obj).hide_viewport();
 }
 
-bool visible_in_collection(LayerCollection* layerCollection, const Object* obj)
-{
+bool visible_in_collection(LayerCollection* lc, const Object* obj) {
     // Check if the object is in the layer collection, if it is, check if the layer is excluded:
-    for (auto collectionObject : blender::list_range((CollectionObject*)layerCollection->collection->gobject.first)) {
+    for (auto collectionObject : blender::list_range((CollectionObject*)lc->collection->gobject.first)) {
         if (collectionObject->ob == obj) {
-            if (!(layerCollection->flag & LAYER_COLLECTION_EXCLUDE)) {
+            if ((!(lc->flag & LAYER_COLLECTION_EXCLUDE)) &&
+#if BLENDER_VERSION >= 300
+                (!(lc->collection->flag & COLLECTION_HIDE_RENDER))
+#else
+                (!(lc->collection->flag & COLLECTION_RESTRICT_RENDER))
+#endif
```

Comment:
what does this change do?

Reply:
This ensures that layer visibility works the same way as object visibility, so meshsync only hides objects that are hidden in render, not viewport.

Reply:
This looks like a separate fix, do you think we could have it in another PR?

Reply:
Done here: https://github.com/Unity-Technologies/MeshSyncDCCPlugins/pull/219/files

Reply:
Please rebase/apply that merged fix to this PR.